sql/catalyst/src/main/scala/org/apache/spark/sql/types/DateUtils.scala

```diff
@@ -39,6 +39,8 @@ object DateUtils {
     millisToDays(d.getTime)
   }
 
+  // TODO(davies): This is buggy, it will be wrong if the date is not aligned with day
+  // we should use the exact day as Int, for example, (year, month, day) -> day
```

Comment:
Please open a jira for it. Also, add an example to explain the problem if you think the `TODO` at here is necessary.